crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -140,19 +137,14 @@ where
                         // Not implemented yet for P2P
                         tracing::info!("Skipping the pending blocks polling");
                         tokio::time::sleep(PENDING_POLL_INTERVAL).await;
-                    } else {
-                        tracing::trace!("Polling pending blocks");
-                        let head =
-                            head_meta.expect("Head hash should exist when entering pending mode");
-                        (next_block, next_state_update) = pending::poll_pending(
+                    } else if pending_handle.is_none() {
+                        tracing::info!("At head of chain, enabling polling of pending data");
```

Comment:
Now that all we do for AtHead is starting the poll_pending task what will keep us from continuously hitting the sequencer with requests for the next block?

Reply:
I've pushed a potential fix for this.

It aggressively polls `head` for changes every 1s since its cheap.

Do you think we need to wrap this one in a timeout?

Reply:
Why is this better than doing a sleep(1) here and then retry the loop?

Reply:
No major reason; I just thought this was easier to reason about than having to know the actual order of events in the rest of the codee. i.e. local reasoning of: this won't proceed until latest changes.

I'm not against doing either.

Reply:
I'm fine with either, too.